src/Kestrel.Core/Internal/Http/HttpHeaders.cs

```diff
@@ -45,7 +45,14 @@ public abstract class HttpHeaders : IHeaderDictionary
                 {
                     ThrowHeadersReadOnlyException();
                 }
-                SetValueFast(key, value);
+                if (value.Count == 0)
```

Comment:
Is this a behavior change?

Reply:
No because it doesn't output null headers; however it means the bit flags can be trusted. Currently if you set the header to a null value it will set the bit flag saying it has a value; rather than changing it to say it doesn't have a value

Reply:
null string/Empty.StringValues/empty array

Reply:
Corner case would be if you set it to an array of null strings; then it would say it had a value with bit flag. However that's same as now (also it needs to know to clear it if its that).

However if you add nulls; StringValues coalesces them to nothing so you'd explicitly need to use the `new StringValues(string[])` constructor

Reply:
With https://github.com/aspnet/Common/pull/323 this changes to
```
value.IsNull
```

Reply:
We should wait for this before merging.

Reply:
Not flowed yet